### PR TITLE
flux-core: add v0.32.0 and v0.33.0

### PR DESCRIPTION
**Pull request by grondo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/flux-core/package.py [hunk 1]
```diff
@@ -45,6 +47,7 @@ class FluxCore(AutotoolsPackage):
     variant('docs', default=False, description='Build flux manpages')
     variant('cuda', default=False, description='Build dependencies with support for CUDA')
 
+    depends_on("ncurses@6.2", when="@0.32.0:,master")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/flux-core/package.py]:
```suggestion
    depends_on("ncurses@6.2", when="@0.32.0:")
```

`Version('master').satisfies('@0.32.0:')`, there's no need to list it separately.

The "special" versions are `develop > main > master > head > trunk > 9999999999`.

File: var/spack/repos/builtin/packages/flux-core/package.py [hunk 2]
```diff
@@ -69,6 +72,7 @@ class FluxCore(AutotoolsPackage):
     depends_on("lz4")
 
     depends_on("asciidoc", type='build', when="+docs")
+    depends_on("py-docutils", type='build', when="@0.32.0:,master")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/flux-core/package.py]: Same here

```suggestion
    depends_on("py-docutils", type='build', when="@0.32.0:")
```

**Review by becker33 — CHANGES_REQUESTED** [review]
Two very minor requests

**Comment by grondo** [comment]
Thanks, @becker33! I believe I have made the requested changes.

**Comment by grondo** [comment]
rebased and force-pushed to re-run CI, the gitlab `build_system-pr-generate` step failed with

```
==> Error: Pipeline generation failed due to the presence of the following specs that are known to be broken in develop:
```

**Comment by grondo** [comment]
Updated with some other minor changes:

 - Remove setting of `FLUX_PMI_LIBRARY_PATH` - which was unnecessary and caused issues on some platforms
 - Fix another spot where `master` was specified unnecessarily
 - update outdated doc link

**Review by becker33 — APPROVED** [review]

**Comment by grondo** [comment]
Thank you!

**Pull request merged**